zkvm/src/types.rs

```diff
@@ -191,6 +199,10 @@ impl Data {
         match self {
```

Comment:
per comment above, let's remove `to_program` entirely and use `to_program_witness` instead where needed, which converts Data::Opaque to PW::Opaque and Data::Program to PW::Program, respectively.